thirdeye/thirdeye-frontend/app/pods/rootcause/route.js

```diff
@@ -1,21 +1,105 @@
 import Ember from 'ember';
+import RSVP from 'rsvp';
+import fetch from 'fetch';
+import moment from 'moment';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
 
 const anomalyRange = [1509044400000, 1509422400000];
 const baselineRange = [1508439600000, 1508817600000];
 const analysisRange = [1508785200000, 1509422400000];
 const urns = new Set(['thirdeye:metric:194591', 'thirdeye:dimension:countryCode:in:provided']);
 
-export default Ember.Route.extend({
-  setupController: function (controller, model) {
+const queryParamsConfig = {
+  refreshModel: false,
+  replace: false
+};
+
+/**
+ * Helper function that checks if a query param 
+ * key/value pair is valid
+ * @param {*} key   - query param key
+ * @param {*} value - query param value
+ * @return {Boolean}
+ */
+const isValid = (key, value) => { 
+  switch(key) {
+    case 'granularity':
+      return ['MINUTES', 'HOURS', 'DAYS'].includes(value);
+    case 'filters':
+      return value && value.length;
+    default:
+      return moment(+value).isValid();
+  }
+};
+
+
+export default Ember.Route.extend(AuthenticatedRouteMixin, {
+  queryParams: {
+    granularity: queryParamsConfig,
+    filters: queryParamsConfig,
+    compareMode: queryParamsConfig,
+    analysisRangeStart: queryParamsConfig,
+    analysisRangeEnd: queryParamsConfig,
+    anomalyRangeStart: queryParamsConfig,
+    anomalyRangeEnd: queryParamsConfig
+  },
+
+  model(params) {
+    const { rootcauseId: id } = params;
+
+    // TODO handle error better
+    if (!id) { return; }
+
+    return RSVP.hash({
+      granularityOptions: fetch(`/data/agg/granularity/metric/${id}`).then(res => res.json()),
+      filterOptions: fetch(`/data/autocomplete/filters/metric/${id}`).then(res => res.json()),
+      maxTime: fetch(`/data/maxDataTime/metricId/${id}`).then(res => res.json()),
```

Comment:
let's not worry about fetching this data yet. This will be handled by asynchronous services.

Reply:
This data needs to be fetched as it prepopulates the data for the rootcause settings component. This is only called once when a route is visited. Let me know how else you'd like to handle this

Reply:
@apucher  passing it down as 'configs' from the controller